drivers/volume/portworx/schedops/k8s-schedops.go

```diff
@@ -227,19 +227,38 @@ func (k *k8sSchedOps) ValidateVolumeSetup(vol *volume.Volume, d node.Driver) err
 		return err
 	}
 
-	nodes := node.GetNodesByName()
-
 	pods, err = k.validatePods(pods, pvName)
 	if err != nil {
 		return err
 	}
 
+	t := func() (interface{}, bool, error) {
+		if err := k.validateMountsInPods(vol, pvName, pods, d); err != nil {
```

Comment:
instead of a retry loop where each retry is for all pods, do it per pod. Here's what could happen. 

You have 10 pods. First 8 are good and pass the check. 9th one fails. In the next retry loop, current code will again check all first 8 pods which is unnecessary.

Reply:
@harsh-px I fixed it, now whenever it has a failure we only act upon the failing pods only, skipping the validated ones